### PR TITLE
hacking guide: use more self-descriptive section headings

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
"Nix with flakes" doesn't thell what the section is about.

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

**Review by iFreilicht — CHANGES_REQUESTED** [review]
Mostly looks good to me. Makes a lot of sense to link the experimental features directly as well.

However, the heading changes break the `#classic-nix` and `#nix-with-flakes` anchors. please fix those. While you're at it, you might as well fix the `#target-platform` anchor as well, maybe by renaming the respective heading to "Target platforms".

**Comment by fricklerhandwerk** [comment]
@iFreilicht done. "Platforms" originally was "Target platforms", but @roberth changed it for undocumented reasons. Should ask before reverting that. I don't have strong opinions on this.

**Comment by roberth** [comment]
>  @roberth changed it for undocumented reasons

Virtually undocumented. The commit message, "Corrections and additions for cross" is not a lot to go by, I have to admit.

Nix is not a compiler, so it does not have a target system, where target is defined as usual by the [GNU docs](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html).

The compiler used to compile Nix does have a target, but that's a contrived way of thinking about a program that's incompatible with how for example Nixpkgs works. It does a good job at hiding target, so if in doubt, you probably shouldn't use the term.

Looking at the first sentence

> Nix can be built for various platforms:

"built for" means the _host_ platform.

If we pretend it's about _target_, that sentence would have to be "Nix can be built to compile to various platforms", which is nonsense.

File: doc/manual/src/contributing/hacking.md [hunk 1]
```diff
@@ -55,20 +56,17 @@ To install it in `$(pwd)/outputs` and test it:
...
 
-To build a release version of Nix:
+To build a release version of Nix for the current operating system and CPU architecture:
 
 ```console
 $ nix build
 ```
 
-You can also build Nix for one of the [supported target platforms](#target-platforms).
+You can also build Nix for one of the [supported target platforms](#platforms).
```
Comment (roberth) [inline doc/manual/src/contributing/hacking.md:65]:
```suggestion
You can also build Nix for one of the [supported platforms](#platforms).
```

File: doc/manual/src/contributing/hacking.md [hunk 2]
```diff
@@ -102,13 +100,13 @@ To install it in `$(pwd)/outputs` and test it:
...
 
-To build Nix for the current operating system and CPU architecture use
+To build a release version of Nix for the current operating system and CPU architecture:
 
 ```console
 $ nix-build
 ```
 
-You can also build Nix for one of the [supported target platforms](#target-platforms).
+You can also build Nix for one of the [supported target platforms](#platforms).
```
Comment (roberth) [inline doc/manual/src/contributing/hacking.md:109]:
```suggestion
You can also build Nix for one of the [supported platforms](#platforms).
```

**Review by roberth — DISMISSED** [review]
Remove target; see earlier comment.

**Comment by roberth** [comment]
We could make cross compilation a lot easier using `default.nix`, as it can take the systems as parameters, fwiw.

**Review by iFreilicht — APPROVED** [review]

**Pull request merged**